Assets/Scripts/Core/GridPathfinding.cs

```diff
@@ -0,0 +1,670 @@
+					while (IsValidShortcut(testStartNode.xPos, testStartNode.yPos, testEndNode.xPos, testEndNode.yPos)) {
+						// Valid shortcut
+						// Remove in between node
+						pathNodeList.RemoveAt(testStartNodeIndex + 1);
+						if (testEndNodeIndex >= pathNodeList.Count - 1) {
+							// No more nodes
+							break;
+						} else {
+							// Test next node
+							testEndNode = pathNodeList[testEndNodeIndex];
+						}
+					}
+					// Start next shortcut test from this end node
+					testStartNodeIndex = testEndNodeIndex;
+				}
+			}
+		}
+
+		public PathRoute GetPathRouteWithShortcuts(Vector3 start, Vector3 end) {
+			List<PathNode> pathNodeList = GetPath(start, end);
+			ApplyShortcuts(ref pathNodeList);
+			return new PathRoute(pathNodeList, worldOrigin_, nodeSize_, null);
+		}
+
+		public PathRoute GetPathRoute(Vector3 start, Vector3 end) {
+			List<PathNode> pathNodeList = GetPath(start, end);
+			return new PathRoute(pathNodeList, worldOrigin_, nodeSize_, null);
+		}
+
+		public List<PathNode> GetPath(int startX, int startY, int endX, int endY) {
+			return findPath(startX, startY, endX, endY);
+		}
+
+		public List<PathNode> GetPath(Vector3 start, Vector3 end) {
+			start = start - worldOrigin_;
+			end = end - worldOrigin_;
+			start = start / nodeSize_;
+			end = end / nodeSize_;
+			MapPos startMapPos = GetClosestValidPos(Mathf.RoundToInt(start.x), Mathf.RoundToInt(start.y));
+			MapPos endMapPos = GetClosestValidPos(Mathf.RoundToInt(end.x), Mathf.RoundToInt(end.y));
+			return findPath(startMapPos.x, startMapPos.y, endMapPos.x, endMapPos.y);
+		}
+
+		public List<PathNode> findPath(int startX, int startY, int endX, int endY) {
+			List<PathNode> ret = new List<PathNode>();
+			// Calculate H for all nodes
+			CalculateAllHeuristics(endX, endY);
+
+			// Start finding target
+			foundTarget_ = false;
+			binaryTree_ = new BinaryTree();
+			openListCount_ = 1;
+
+			PathNode currentNode = mapNodes_[startX][startY];
+			PathNode targetNode = mapNodes_[endX][endY];
+
+			if (currentNode == targetNode) {
+				return new List<PathNode> { currentNode };
+			}
+			int iterations = 0;
+			do {
+				iterations++;
+				currentNode = FindTarget(currentNode, targetNode);
+			} while (!foundTarget_ && openListCount_ > 0 && iterations < 60000);
+			if (iterations >= 60000) UnityEngine.Debug.Log("iteration overload");
+
+			if (foundTarget_) {
+				// Get path
+				currentNode = targetNode;
+				ret.Add(currentNode);
+				while (currentNode.parent != null && currentNode.parent != currentNode) {
+					ret.Add(currentNode.parent);
+					currentNode = currentNode.parent;
+				}
+				if (currentNode.parent == currentNode) UnityEngine.Debug.Log("parent == child");
+			} else {
+				// No path possible
+			}
+			ret.Reverse();
+
+			return ret;
+		}
+
+		private PathNode FindTarget(PathNode currentNode, PathNode targetNode) {
+			// Check the north node
+			if (currentNode.moveNorth) DetermineNodeValues(currentNode, currentNode.north, targetNode);

```

Comment:
Coding style: single line bodies of your if-statement go to a new line (this is the same for the follow three if-statements as well)